dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/ClassDumper.java

```diff
@@ -76,13 +87,118 @@
       ConstantNameAndType constantNameAndType = (ConstantNameAndType) constantAtNameAndTypeIndex;
       String methodName = constantNameAndType.getName(constantPool);
       String descriptor = constantNameAndType.getSignature(constantPool);
-      FullyQualifiedMethodSignature methodref = new FullyQualifiedMethodSignature(className,
-          methodName, descriptor);
+      FullyQualifiedMethodSignature methodref =
+          new FullyQualifiedMethodSignature(classNameInMethodReference, methodName, descriptor);
       methodReferences.add(methodref);
     }
     return methodReferences;
   }
 
+  /**
+   * Lists external method references for a class. The returned list does not include method
+   * references that point to other classes defined in the same jar file as the class.
+   *
+   * @param className class name to list its method references. The class must be available through
+   *     the class loader and BCEL repository.
+   * @param jarFileToClasses mapping of jar file paths to classes. This helps to distinguish whether
+   *     the class in a method reference is in the same jar file or not.
+   * @param classLoader class loader to locate the jar file for the class
+   * @param repository BCEL repository to list method references for the class
+   * @return list of external method references from the class
+   */
+  static List<FullyQualifiedMethodSignature> listExternalMethodReferences(
+      String className,
+      SetMultimap<Path, String> jarFileToClasses,
+      ClassLoader classLoader,
+      SyntheticRepository repository) {
+    try {
+      JavaClass javaClass = repository.loadClass(className);
```

Comment:
Can you move this down to where you need it? It helps readability to minimize the visible scope of variables. You don't use `javaClass` until line 124. If you move it there then I don't have to keep track of the difference between `javaClass` and `clazz` for as long.

Better yet, I _think_ you only use `javaClazz` one time. If you inline it into that usage, the reader never has to think about it.